vignettes/articles/writing-custom-fns.Rmd

```diff
@@ -0,0 +1,244 @@
+
+- `check_file_*` for checks that operate on model output files (e.g. file location, name etc).
+- `check_tbl_*` for checks that operate on a tibble of model output data (i.e. the contents of a file).
+- `check_meta_*` for checks that operate on model metadata files.
+- `check_submission_*` for checks that operate on high level properties of the submission (e.g. timing).
+- `check_config_*` for checks that operate on the hub configuration files.
+- `check_valid_*` for checks that don't fit into the above categories.
+
+To distinguish `hubValidations` package functions from custom ones, we also recommend prefixing custom function names with an additional prefix, e.g. `cstm_` or `cs`.
+
+</div>
+
+## Arguments to custom check functions
+
+### Arguments available in the caller environment 
+
+The `hubValidations` package automatically passes a set of standard objects to custom check functions. These objects are passed to arguments of custom check functions with the same name and do not need to be included in the `args` configuration in the `validations.yml` file.
+
+
+```{r child="children/_custom-fn-available-args.Rmd", echo=FALSE, results="asis"}
+```
+
+### Additional arguments
+
+You can add additional arguments to custom check functions and pass values to them by including them in the `args` configuration in the `validations.yml` file. These values are passed to the custom check function by `hubValidations` when the function is called.
+
+
+If you do add additional arguments to a custom check function, you should also add input  checks at the start of the function to ensure inputs are valid. Package `checkmate` contains a wide range of functions for checking inputs.
+
+For example, the optional check `opt_check_tbl_col_timediff()` (which is deployed in exactly the same fashion as custom functions, i.e. through the `validations.yml` file) takes additional arguments `t0_colname`, `t1_colname` and `timediff`. 
+
+
+```{r}
+opt_check_tbl_col_timediff
+```
+
+
+
+You can add an example extra argument with `extra_args = TRUE` when creating the custom check function with `create_custom_check()`.
+
+```{r}
+create_custom_check("cstm_check_tbl_args",
+  hub_path = hub_path,
+  extra_args = TRUE
+)
+```
+
+This adds an extra argument `extra_arg` to the custom check function formals as well as an example input check.
+s
+```{r, code=readLines(r_path(hub_path, "cstm_check_tbl_args")), eval=FALSE}
+```
+
+
+## Function output
+
+### Capturing and returning check results
+
+**Returning a check condition with `capture_check_cnd()`**
+
+The `capture_check_cnd()` function is used to return a check condition and it's output should also be what a custom check function returns in most cases. The function returns a `<hub_check>` class object depending on the value passed to the `check` argument, which represents the summary of the condition being checked by a given validation function. 
```

Comment:
I'm keeping the "n most cases" because the following section clearly describes a situation where the output of `capture_check_cnd()` is not what the function returns

Reply:
I am ok with keeping the `in most cases`, I was just not clear what it meant. The sentence was not clear to me, but I like the new version, it's all clear to me now! thank you again